### PR TITLE
Use the hostname and port for GraphQL connections

### PR DESCRIPTION
**Pull request by shicholas**
Addresses https://github.com/PyGithub/PyGithub/issues/2870

@EnricoMi, I noticed that there are some ConnectionClasses, is that what you had envisioned? should the `requestJsonAndCheck` method be moved to those classes?

Thanks for this awesome lib!

### CODE REVIEW

File: github/Requester.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def graphql_query(self, query: str, variables: Dict[str, Any]) -> Tuple[Dict[str
         """
         input_ = {"query": query, "variables": {"input": variables}}
 
-        response_headers, data = self.requestJsonAndCheck("POST", "https://api.github.com/graphql", input=input_)
+        response_headers, data = self.requestJsonAndCheck("POST", f"https://{self.__hostname}:{self.__port}/graphql", input=input_)
```
Comment (EnricoMi) [inline github/Requester.py:543]: Right, that is easier than I thought, but that should do it.

I'd suggest to construct the GraphQl URL in the `__init__` method right after `o = urllib.parse.urlparse(base_url)` (line 397) like this:
```
o = urllib.parse.urlparse(base_url)
self.__graphql_url = f"{o.scheme}://{o.hostname}:{o.port}/graphql"
```

or maybe safer would be this
```
o = urllib.parse.urlparse(base_url)
self.__graphql_url = urlunparse(o._replace(path="graphql"))
```

And then use `self.__graphql_url` here instead:
```suggestion
        response_headers, data = self.requestJsonAndCheck("POST", self.__graphql_url, input=input_)
```

**Comment by EnricoMi** [comment]
We could use some test here.

File: github/Requester.py (hunk 1 above)
Comment (EnricoMi) [reply github/Requester.py:543]: Above suggestion using `urlunparse` should also fix the test error `ValueError: Port could not be cast to integer value as 'None'`.

**Comment by EnricoMi** [comment]
I have fixed the imports...

**Comment by EnricoMi** [comment]
I have added a few tests. Please test the code against your enterprise server.

**Comment by EnricoMi** [comment]
What is the path of your v3 REST API? Is it `/api/v3`?

**Comment by EnricoMi** [comment]
If you don't want the attribution, I have to do this without your commits: #2880

**Comment by EnricoMi** [comment]
Fixed in #2880.

**Comment by EnricoMi** [comment]
Fixed in #2880.
